src/odemis/gui/cont/tabs.py

```diff
@@ -7202,6 +7279,139 @@ def _on_autofocus_done(self, future):
+        # assign the button which got the click event
+        btn = event.GetEventObject()
+
+        # GUI stream bar controller pauses the stream
+        self._stream_controller.pauseStreams()
+
+        if btn.GetLabel() in {"Engage", "Engaged"}:
+            # set the requested position to ACTIVE and change the appearance of the retract and engage buttons
+            pos = spec_switch[model.MD_FAV_POS_ACTIVE]
+            btn.SetLabel("Cancel")
+            self.panel.btn_spec_switch_retract.SetForegroundColour(wx.BLACK)
+            self.panel.btn_spec_switch_retract.SetLabel("Retract")
+        elif btn.GetLabel() in {"Retract", "Retracted"}:
+            # set the requested position to DEACTIVE and change the appearance of the retract and engage buttons
+            pos = spec_switch[model.MD_FAV_POS_DEACTIVE]
+            btn.SetLabel("Cancel")
+            self.panel.btn_spec_switch_engage.SetForegroundColour(wx.BLACK)
+            self.panel.btn_spec_switch_engage.SetLabel("Engage")
+        elif btn.GetLabel() == "Cancel":
+            # stop the current movement of the actuator and force the button_done callback
+            main.spec_switch.stop()
+            self._on_specswitch_button_done(event)
+            return
+
+        # disable the alignment buttons when moving the mirror
+        self.panel.btn_m_spec_switch_x.Enable(False)
+        self.panel.btn_p_spec_switch_x.Enable(False)
+
+        # show the gauge if it is hidden
+        if not self.panel.gauge_specswitch.IsShown():
+            self.panel.gauge_specswitch.Show()
+
+        # unsubscribe to position changes to prevent overwriting FAV_POS
+        if self.tab_data_model.main.spec_switch:
+            self.tab_data_model.main.spec_switch.position.unsubscribe(self._onSpecSwitchPos)
+
+        # move the mirror to the right position
+        self._specswitch_f = main.spec_switch.moveAbs(pos)
+        self._specswitch_f.add_done_callback(self._on_specswitch_button_done)
+        # take care of a moving gauge while moving the mirror
+        self._gauge_f = self.move_specswitch_mirror(pos)
+        self._pfc_specswitch = ProgressiveFutureConnector(self._gauge_f, self.panel.gauge_specswitch)
+
+    def move_specswitch_mirror(self, pos: dict) -> model.ProgressiveFuture:
+        """
+        Handles progression updates of the gauge in a  future which is
+        calculated from the start of the move until the end of the move.
+        :param pos: the requested position of the spec-switch mirror.
+        :return: progressive future controlling the gauge update
+        """
+        spec_switch = self.tab_data_model.main.spec_switch
+        est_start = time.time() + 0.1
+
+        # calculate the time the mirror needs to move from the current position to the requested position
+        current_pos = spec_switch.position.value
+        current_speed = spec_switch.speed.value
+        mirror_move_time = abs((current_pos.pop('x') - pos.pop('x')) / current_speed.pop('x'))
```

Comment:
I don't have a clue anymore as to why I used pop instead of directly accessing through the key.. so I changed it :)
Also added a fallback for the speed.